addons/web/static/src/js/view_form.js

```diff
@@ -4507,7 +4507,11 @@ instance.web.form.One2ManyListView = instance.web.ListView.extend({
             var cached_records = _.any([ds.to_create, ds.to_delete, ds.to_write], function(value) {
                 return value.length;
             });
-            if (!self.o2m.options.reload_on_button && !cached_records) {
+            if (self.o2m.options.reload_whole_on_button && !cached_records){
+                self.handle_button(name, id, function(){
+                self.o2m.dataset.parent_view.recursive_reload();
+                });
+            }else if (!self.o2m.options.reload_on_button && !cached_records) {
```

Comment:
This is the kind of commits that turn our codebase into a terrifying spaghetti monster.  Rim is totally right:
1. it should be done in a separate commit with an explanation.
2. The "options" is not a bag that you can fill with anything.  Please think about your needs, and if needed, the list view can be updated by adding a method or something.  But this is not the way to do it.  This is actually the opposite of OOP.  We want to delegate the specific details of what needs to be done to the object concerned with them, not delegate the details to the most generic object and add a bunch of specific cases.